polaris/hub/client.py

```diff
@@ -458,120 +490,133 @@ def upload_dataset(
+            dataset.owner = HubOwner.normalize(owner or dataset.owner)
+            dataset_json = dataset.model_dump(
+                exclude={"cache_dir", "table"}, exclude_none=True, by_alias=True
             )
 
-        # Normalize timeout
-        if timeout is None:
-            timeout = self.settings.default_timeout
-
-        # Get the serialized data-model
-        # We exclude the table as it handled separately and we exclude the cache_dir as it is user-specific
-        dataset.owner = HubOwner.normalize(owner or dataset.owner)
-        dataset_json = dataset.model_dump(exclude={"cache_dir", "table"}, exclude_none=True, by_alias=True)
-
-        # If the dataset uses Zarr, we will save the Zarr archive to the Hub as well
-        if dataset.uses_zarr:
-            dataset_json["zarrRootPath"] = f"{PolarisFileSystem.protocol}://data.zarr"
-
-        # Uploading a dataset is a three-step process.
-        # 1. Upload the dataset meta data to the hub and prepare the hub to receive the data
-        # 2. Upload the parquet file to the hub
-        # 3. Upload the associated Zarr archive
-        # TODO: Revert step 1 in case step 2 fails - Is this needed? Or should this be taken care of by the hub?
-
-        # Prepare the parquet file
-        buffer = BytesIO()
-        dataset.table.to_parquet(buffer, engine="auto")
-        parquet_size = len(buffer.getbuffer())
-        parquet_md5 = md5(buffer.getbuffer()).hexdigest()
-
-        # Step 1: Upload meta-data
-        # Instead of directly uploading the data, we announce to the hub that we intend to upload it.
-        # We do so separately for the Zarr archive and Parquet file.
-        url = f"/dataset/{dataset.artifact_id}"
-        response = self._base_request_to_hub(
-            url=url,
-            method="PUT",
-            json={
-                "tableContent": {
-                    "size": parquet_size,
-                    "fileType": "parquet",
-                    "md5Sum": parquet_md5,
+            # If the dataset uses Zarr, we will save the Zarr archive to the Hub as well
+            if dataset.uses_zarr:
+                dataset_json["zarrRootPath"] = f"{PolarisFileSystem.protocol}://data.zarr"
+
+            # Uploading a dataset is a three-step process.
+            # 1. Upload the dataset meta data to the hub and prepare the hub to receive the data
+            # 2. Upload the parquet file to the hub
+            # 3. Upload the associated Zarr archive
+            # TODO: Revert step 1 in case step 2 fails - Is this needed? Or should this be taken care of by the hub?
+
+            # Prepare the parquet file
+            buffer = BytesIO()
+            dataset.table.to_parquet(buffer, engine="auto")
+            parquet_size = len(buffer.getbuffer())
+            parquet_md5 = md5(buffer.getbuffer()).hexdigest()
+
+            # If the dataset uses Zarr, we will save the Zarr archive to the Hub as well
```

Comment:
Is this duplicated? Looks like it appears on L515 as well.

Reply:
Yes, I think it snuck in when resolving some merge conflicts. Great catch.